src/main/java/org/candlepin/model/ContentCurator.java

```diff
@@ -645,63 +646,65 @@ public Set<Content> getChildrenContentOfProductsByUuids(Collection<String> produ
+                JOIN cp_contents content ON content.uuid = pc.content_uuid
+            """;
+
+        Map<String, Boolean> contentEnablementMap =
+            (Map<String, Boolean>) entityManager.createNativeQuery(sql)
             .setParameter("owner_id", ownerId)
             .setParameter("start_date", now)
             .setParameter("end_date", now)
-            .getResultList());
-
-        Query prodQuery = entityManager.createNativeQuery(sql);
-        int blockSize = Math.min(this.getInBlockSize(), this.getQueryParameterLimit() / 2);
-
-        // Get all of the derived products and provided products for the products we've found, recursively
-        for (Set<String> lastBlock = productUuids; !lastBlock.isEmpty();) {
-            Set<String> children = new HashSet<>();
-
-            for (List<String> block : this.partition(lastBlock, blockSize)) {
-                children.addAll(prodQuery.setParameter("product_uuids", block)
-                    .getResultList());
-            }
-
-            productUuids.addAll(children);
-            lastBlock = children;
-        }
-
-        // Use the product UUIDs to select all related enabled content
-        jpql = "SELECT DISTINCT pc.content, pc.enabled FROM ProductContent pc " +
-            "WHERE pc.product.uuid IN (:product_uuids)";
-
-        Query contentQuery = entityManager.createQuery(jpql);
-        HashMap<Content, Boolean> activeContent = new HashMap<>();
-
-        for (List<String> block : this.partition(productUuids)) {
-            contentQuery.setParameter("product_uuids", block)
-                .getResultList()
-                .forEach(col -> {
-                    Content content = (Content) ((Object[]) col)[0];
-                    Boolean enabled = (Boolean) ((Object[]) col)[1];
-                    activeContent.merge(content, enabled, (v1, v2) ->
-                        v1 != null && v1.booleanValue() ? v1 : v2);
-                });
+            .getResultList()
+            .stream()
+            .collect(Collectors.toMap(
+                row -> (String) ((Object[]) row)[0],  // UUID of the content
+                row -> (Boolean) ((Object[]) row)[1], // Enabled state
+                (enabled1, enabled2) -> enabled1 || enabled2 // Collision resolution: logical OR
+            ));
+
+        List<ProductContent> activeContent = new ArrayList<>();
+
+        String contentLookupJPQL =
+            "SELECT content FROM Content content WHERE content.uuid IN (:content_uuids)";
+        TypedQuery<Content> contentQuery = entityManager.createQuery(contentLookupJPQL, Content.class);
+
+        // This step is necessary to handle large sets of UUIDs and avoid query parameter limits
+        for (List<String> block : this.partition(contentEnablementMap.keySet())) {
+            contentQuery.setParameter("content_uuids", block);
+            List<Content> contents = contentQuery.getResultList();
+            contents.stream()
```

Comment:
Same with this. The whole operation should be chainable here from contentQuery down:

```
contentQuery.setParam(...)
  .getResultStream()
  .map(...)
  ...
```

Reply:
Answer above and done